internal/handlers/common/saslstart_plain_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"encoding/base64"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+// TestSaslStartPlain tests saslStartPlain function.
+// Integration tests are not possible because the driver
+// used in integration tests doesn't support all possible scenarios.
+// To ensure compatibility, the functionality must be tested by external drivers
+// of various programming languages (see the dance repo).
+func TestSaslStartPlain(t *testing.T) {
+	validPayload := []byte{
+		97, 117, 116, 104, 122, 105, 100, // "authzid"
```

Comment:
Why not just use `[]byte("admin")` or `p = append(p, "admin"...)` or something? Now we should keep numbers and strings in sync manually